meterfs/meterfs.c

```diff
@@ -305,11 +312,255 @@ static int meterfs_eraseFileTable(unsigned int n, meterfs_ctx_t *ctx)
 }
 
 
+static int meterfs_isHeaderValid(unsigned int headerIdx, index_t *id, unsigned int *filecnt, bool *mustMigrate, meterfs_ctx_t *ctx)
+{
+	ssize_t err;
+	union {
+		header_t h;
+		fileheader_t f;
+		headerOld_t hOld;
+		fileheaderOld_t fOld;
+	} u;
```

Comment:
Consider splitting to functions with specific field on stack. This way you can avoid mistakes of using shared stack memory errors.